app/src/processing/app/Base.java

```diff
@@ -1360,6 +1367,39 @@ public void onBoardOrPortChange() {
       }
     }
 
+    // Update recent boards list in preferences
+    List<String> newRecentBoardIds = new ArrayList<String>();
+    String currentBoard = PreferencesData.get("board");  
+    for (String recentBoard : PreferencesData.getCollection("recent.boards")){
+      if (!recentBoard.equals(currentBoard)) {
+        newRecentBoardIds.add(recentBoard);
+      }
+    }
+    newRecentBoardIds.add(0, currentBoard);
+
+    int numBoards = 0;
+
+    if (PreferencesData.has("recent.num_boards")) {
+      numBoards = PreferencesData.getInteger("recent.num_boards");
+    }
```

Comment:
`getInteger` can take a default argument, so this can be:

```suggestion
    int numBoards = PreferencesData.getInteger("recent.num_boards", 0);
```

But, given what I say about the default `preferences.txt` in another comment, I think the default can even be left out:

```suggestion
    int numBoards = PreferencesData.getInteger("recent.num_boards");
```

Reply:
I did this in my branch, as part of some other cleanup.